modules/nextflow/src/main/groovy/nextflow/trace/GraphObserver.groovy

```diff
@@ -81,6 +82,12 @@ class GraphObserver implements TraceObserver {
     void onFlowComplete() {
         // -- normalise the DAG
         dag.normalize()
+
+        // -- make sure parent path exists
+        final parent = file.getParent()
+        if( parent )
+            Files.createDirectories(parent)
```

Comment:
```suggestion
        file.parent?.mkdirs()
```

Nit ☝️

Reply:
I just copied the code from the other report observers. Do you want to update those too?

Reply:
As you prefer

Reply:
Let's just leave the others as they are

Reply:
If accepting this change, as it looks it happened, then the extra added import (for `java.nio.file.Files`) is unnecessary.

Reply:
You're right that it's unnecessary, but I think having it doesn't hurt either